mozilla/foundation.mozilla.org
# [7435] add tito ticket complete webhook

Closes #7435 

**Link to sample test page**: https://mozfest-foundation-s-7435-featu-ixc3yh.mofostaging.net/en/tickets/

Steps to test:

1. Go to page, click Get Tickets and fill out payment form
2. Choose to allocate ticket and click the checkbox for "_Check the box below to be added to our Mozilla Festival list and be the first to know about the latest news and updates from the festival. You'll also be joining our Mozilla news list to stay informed in the fight for a better internet for all._"
3. Submit the form and check email for "Action Required: final step" email. I think it'll only be sent once per email address. See screenshot below for example.

Caveats:

1. Due to time constraints this PR does not include tests, documentation or logging. Also, the J[S](https://github.com/mozilla/foundation.mozilla.org/pull/7938/files) added for the previous Tito ticket needs to be removed. If this PR is approved I'll create a separate ticket for this.
2. Ideally the code would be refactored with less coupling, and include error handling for the Basket API call.
3. This covers the use case where the user completes a ticket and signs-up for the newsletter. It does not cover the potential use case where someone edits their ticket and unchecks the sign-up.
4. It only caters for one Tito event at a time, and assumes there is only one field for signing up to a newsletter. Support for multiple Tito events will require more work.


## Checklist

**Tests**

**Documentation:**




![Screenshot from 2021-12-06 19-57-11](https://user-images.githubusercontent.com/3485411/144913514-31849bf9-5a53-4d56-9bf4-2b7222761eff.png)
